components/server/src/jobs/workspace-gc.ts

```diff
@@ -70,7 +72,7 @@ export class WorkspaceGarbageCollector implements Job {
                 );
             const afterSelect = new Date();
             const deletes = await Promise.all(
-                workspaces.map((ws) => this.deletionService.softDeleteWorkspace({ span }, ws, "gc")),
+                workspaces.map((ws) => this.workspaceService.deleteWorkspace(ws.ownerId, ws.id, "gc")), // TODO(gpl) This should be a system user/service account instead of ws owner
```

Comment:
This would work, but is not correct as this is not triggered by the owner but by the system. 
In another place (`UserService#findById`) I have used `undefined` as the userId to indicate the request is coming from the system.

Reply:
Yeah, I'm not particularly satisfied with that choice for the protocol. But I liked it better than introducing a system ID or anything else I could think of. 🤷

Reply:
[`UserService#findById` ](https://github.com/gitpod-io/gitpod/blob/9b4cc8507d86ce70a74d373a473456015c7cf6cd/components/server/src/iam/iam-session-app.ts#L150)

Reply:
After looking at the change, I really thing we should make this a separate argument, instead of overriding userId for the security reasons. Let's discuss in sync. :+1: